examples/chip-tool/commands/pairing/PairingCommand.cpp

```diff
@@ -439,3 +452,20 @@ void PairingCommand::OnAddressUpdateComplete(NodeId nodeId, CHIP_ERROR err)
         SetCommandExitStatus(err);
     }
 }
+
+void PairingCommand::OnDiscoveredDevice(const chip::Mdns::DiscoveredNodeData & nodeData)
+{
+    const uint16_t port = nodeData.port;
+    char buf[chip::Inet::kMaxIPAddressStringLength];
+    nodeData.ipAddress[0].ToString(buf);
+    ChipLogProgress(chipTool, "Discovered Device: %s:%u", buf, port);
+
+    // Stop Mdns discovery. Is it the right method ?
+    GetExecContext()->commissioner->RegisterDeviceDiscoveryDelegate(nullptr);
+
+    CHIP_ERROR err = Pair(mRemoteId, PeerAddress::UDP(nodeData.ipAddress[0], port, nodeData.interfaceId[0]));
```

Comment:
This is probably not quite right.  We only want to use the discovered interface id if the ip tests true for `IsIPv6LinkLocal`... Otherwise we might run into problems because the discovery happened on on interface but the actual communication in fact needs to be on a different interface.

See the long comment in `DeviceController::OnNodeIdResolved`.

Probably OK to fix this in a followup, but we should just have a helper for this....

Reply:
Fixed.